st2client/st2client/models/core.py

```diff
@@ -436,6 +436,22 @@ def resume(self, execution_id, **kwargs):
 
         return self.resource.deserialize(response.json())
 
+    @add_auth_token_to_kwargs_from_env
+    def get_children(self, execution_id, **kwargs):
+        url = '/%s/%s/children' % (self.resource.get_url_path_name(), execution_id)
+
+        depth = kwargs.pop('depth', -1)
+
+        params = kwargs.pop('params', {})
+
+        if depth:
+            params['depth'] = depth
+
+        response = self.client.get(url=url, params=params, **kwargs)
+        if response.status_code != http_client.OK:
+            self.handle_error(response)
+        return [self.resource.deserialize(item) for item in response.json()]
```

Comment:
deserialize could potentially through. So a list comprehension may not be the best approach here. Since this is a client facing tool, we should try to be more precise about the error and not raise just the underlying exception all the way. 

Maybe it's best to build the list manually and on exception in a single item, raise an exception with a message like "Invalid resource format from server for item %s  in response %s" % (item, response.json()) might be better?

Reply:
Does manually constructing the list differ from just wrapping the comp in a try?
```suggestion
        try:
            return [self.resource.deserialize(item) for item in response.json()]
        except JSONDecodeError as e:
            raise JSONDecodeError("Invalid resource format from server for item %s in response %s" % (item, response.json()))
```
A similar list comp appears elsewhere in the same file, the `get_by_id()` and `get_property()` methods of `ResourceManager` implement them.  Should they be updated to catch this error as well?